model/app.py

```diff
@@ -411,14 +411,18 @@ def fix_measurements_ids(measurements):
 ReactionKnockouts = namedtuple('ReactionKnockouts', ['model', 'changes'])
 
 
-async def operate_on_reactions(model, reactions_ids, key, apply_function, undo_function):
+async def operate_on_reactions(model, reactions, key, apply_function, undo_function):
     if 'changes' not in model.notes:
         model.notes['changes'] = deepcopy(EMPTY_CHANGES)
     current = model.notes['changes'][key]['reactions']
     applied = set([r['id'] for r in current])
-    to_apply = set(reactions_ids) - applied
-    to_undo = [r for r in current
-               if r['id'] in (applied - set(reactions_ids))]
+    try:
```

Comment:
that's a hacky way to check if the reactions represented as ids or as dictionary with id and string: before introducing strings knockouts and additions were processed in uniformed way. I see it should be changed to check for the operation key instead of the error raised, so it's more readable